lightwood/encoders/numeric/ts_numeric.py

```diff
@@ -0,0 +1,128 @@
+import math
+import sys
+
+import torch
+import numpy as np
+from lightwood.encoders.numeric import NumericEncoder
+from lightwood.logger import log
+
+
+class TsNumericEncoder(NumericEncoder):
+    """
+    Variant of vanilla numerical encoder, supports dynamic mean re-scaling
+    """
+    def __init__(self, is_target=False):
+        super(TsNumericEncoder, self).__init__(is_target=is_target)
+        # time series normalization params
+        self.normalize_by_group = False
+        self.normalizers = None
+        self.group_combinations = None
```

Comment:
what does group_combinations refer to

Reply:
Good point, I'll add a comment. `group_combinations` are all sets of grouped by columns observed at training time.

Imagine we group by 2 columns: `country` and `size`. If the possible values for these in the training corpus are, respectively, `USA, UK` and `Big, Medium, Small`, then `group_combinations` will hold all 6 possible combinations between these two sets. 

We use a `frozenset()` of the respective strings to represent these combinations, so that we can retrieve normalizer objects at inference time once we know what combination we are looking at.